builders/xcm/fees.md

```diff
@@ -0,0 +1,230 @@
+
+Substrate has introduced a weight system that determines how heavy an extrinsic is. When it comes to paying fees, users will pay a transaction fee that is proportionate to the weight of the call that is being made. One unit of weight is defined as one picosecond of execution time.
+
+The following sections will break down how to calculate XCM fees for Polkadot, Kusama, and Moonbeam-based networks. It's important to note that Kusama uses benchmarking and database read and write weight costs to determine the total weight costs for XCM instructions. Whereas Polkadot and Moonbeam-based networks use a fixed weight cost in their calculations. 
+
+There are two databases available in Polkadot and Kusama, RocksDB (which is the default) and ParityDB, both of which have their own associated weight costs for each network. 
+
+### Polkadot {: #polkadot }
+
+As previously mentioned, Polkadot currently uses a fixed amount of weight for all XCM instructions, which is `1,000,000,000` weight units.
+
+Although Polkadot doesn't currently use database weight units to calculate costs, the weight units for database operations are shared here for reference.
+
+|     Database      |    Read    |   Write    |
+|:-----------------:|:----------:|:----------:|
+| RocksDB (default) | 20,499,000 | 83,471,000 |
+|     ParityDB      | 11,826,000 | 30,052,000 |
+
+With the instruction weight cost established, you can calculate the cost of the instruction in DOT. 
+
+In Polkadot, the `ExtrinsicBaseWeight` is set to `85,212,000` which is mapped to 1/10 cents. Where 1 cent is `10^10 / 10,000`. Therefore, the formula for calculating the final fee in DOT looks like this:
+
+```
+Total-Planck-DOT = TotalWeight * PlanckDOT * (1 / ExtrinsicBaseWeight)
+DOT = Total-Planck-DOT / 10^10
+```
+
+Therefore, the actual calculation for the `WithdrawAsset` instruction is:
+
+```
+Total-Planck-DOT = 1000000000 * (10^10 / 1000) * (1 / 85212000)
+DOT = 117354360 / 10^10
+```
+
+The total cost is `0.0117354363 DOT`.
+
+### Kusama {: #kusama }
+
+Kusama doesn't used fixed weights, but bases the weights off of measured results from benchmarking data. The weights on Kusama take into consideration database reads and writes in addition to the weight required for a given instruction. 
+
+The breakdown of weight costs for database read and write operations are as follows:
+
+|     Database      |    Read    |    Write    |
+|:-----------------:|:----------:|:-----------:|
+| RocksDB (default) | 25,000,000 | 100,000,000 |
+|     ParityDB      | 8,000,000  | 50,000,000  |
+
+Now that you are aware of the weight costs for database reads and writes on Kusama, you can calculate the weight cost for a given instruction using the base weight for an instruction. 
+
+The `WithdrawAsset` instruction has a base weight of `20,385,000`, and includes one database read, and one database write. Therefore, the total weight cost of the `WithdrawAsset` instruction is calculated like so:
```

Comment:
Add URL for reference: https://github.com/paritytech/polkadot/blob/master/runtime/kusama/src/weights/xcm/pallet_xcm_benchmarks_fungible.rs#L49-L53